src/main/java/org/spongepowered/common/mixin/core/network/rcon/MixinRConConsoleSource.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.network.rcon;
+
+import net.minecraft.command.ICommandSender;
+import net.minecraft.network.rcon.RConConsoleSource;
+import net.minecraft.network.rcon.RConThreadClient;
+import net.minecraft.util.IChatComponent;
+import org.spongepowered.api.net.RemoteConnection;
+import org.spongepowered.api.service.permission.PermissionService;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.Tristate;
+import org.spongepowered.api.util.command.source.RconSource;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.interfaces.IMixinRConConsoleSource;
+import org.spongepowered.common.interfaces.Subjectable;
+import org.spongepowered.common.text.SpongeTexts;
+
+@Mixin(RConConsoleSource.class)
+public abstract class MixinRConConsoleSource implements ICommandSender, IMixinRConConsoleSource, RconSource, Subjectable {
+    @Shadow
+    private StringBuffer buffer;
+
+    private RConThreadClient clientThread;
+
+    @Override
+    public RemoteConnection getConnection() {
+        return (RemoteConnection) this.clientThread;
+    }
+
+    @Override
+    public void setConnection(RConThreadClient conn) {
+        this.clientThread = conn;
+    }
+
+    @Override
+    public void setLoggedIn(boolean loggedIn) {
+        this.clientThread.loggedIn = loggedIn;
+    }
+
+    @Override
+    public boolean getLoggedIn() {
+        return this.clientThread.loggedIn;
+    }
+
+    @Override
+    public void sendMessage(Iterable<Text> messages) {
+        for (Text message : messages) { // TODO: Locale?
+            this.addChatMessage(SpongeTexts.toComponent(message));
+        }
+    }
+
+    @Override
+    public void sendMessage(Text... messages) {
+        for (Text message : messages) {
+            this.addChatMessage(SpongeTexts.toComponent(message));
+        }
+    }
+
+    /**
+     * Add newlines between output lines for a command
+     * @param component
+     */
+    @Inject(method = "addChatMessage", at = @At("RETURN"))
+    public void addNewlines(IChatComponent component, CallbackInfo ci) {
+        this.buffer.append('\n');
+    }
+
+    @Override
+    public String getName() {
+        return getCommandSenderName();
```

Comment:
should qualify `this` at all times, I recommend  making your IDE automatically do it.